src/shared/sam/localLambdaRunner.ts

```diff
@@ -668,6 +668,7 @@ async function showDebugConsole(): Promise<void> {
     }
 }
 
+// TODO: rework this function. The fact that it is mutating the config file AND writing to disk is bizarre.
```

Comment:
Just for reference, it was an incremental improvement vs the previous state, where similar logic was duplicated in 3+ different places. The gain here is that (1) now the logic is isolated in this one, ugly place :) and (2) we have a strong separate between "CONFIG" state and "EXECUTE" stage. 

(The "CONFIG" stage covers most of the complexity, so the EXECUTE stage is minimal. So we can test the state of the CONFIG stage, with many fast tests instead of slow integ tests.)

Reply:
This function is pretty compact. Trying to deal with these mutations in 3 places sounds rough.